client/my-sites/checkout/checkout-thank-you/jetpack-plan-details.jsx

```diff
@@ -4,85 +4,133 @@
-					'installation by following {{a}}these instructions{{/a}}.',
-				{
-					components: {
-						a: (
-							<a
-								target="_blank"
-								rel="noopener noreferrer"
-								href="https://en.support.wordpress.com/setting-up-premium-services/"
-								onClick={ trackManualInstall( 'calypso_plans_autoconfig_click_opt_out' ) }
-							/>
-						),
-					},
-				}
+			return (
+				<PurchaseDetail
+					icon="cog"
+					buttonText={ translate( 'Installation Instructions' ) }
+					description={ translate(
+						'You can now install Akismet and VaultPress, which will automatically ' +
+							'protect your site from spam and data loss. ' +
+							"If you have any questions along the way, we're here to help!"
+					) }
+					href="https://en.support.wordpress.com/setting-up-premium-services/"
+					onClick={ trackManualInstall }
+				/>
 			);
-			props.buttonText = i18n.translate( 'Set up your plan' );
-			props.href = `/plugins/setup/${ selectedSite.slug }`;
 		}
+
+		return (
+			<PurchaseDetail
+				icon="cog"
+				buttonText={ translate( 'Set up your plan' ) }
+				description={ translate(
+					'We are about to install Akismet and VaultPress for your site, ' +
+						'which will automatically protect your site from spam ' +
+						'and data loss. If you have any questions along the way, ' +
+						"we're here to help! You can also perform a manual " +
+						'installation by following {{a}}these instructions{{/a}}.',
+					{
+						components: {
+							a: (
+								<a
+									target="_blank"
+									rel="noopener noreferrer"
+									href="https://en.support.wordpress.com/setting-up-premium-services/"
+									onClick={ trackManualInstall }
+								/>
+							),
+						},
+					}
+				) }
+				href={ `/plugins/setup/${ selectedSite.slug }` }
+			/>
+		);
+	}
+}
+
+const getTracksDataForAutoconfigHalt = selectedSite => {
+	const reasons = utils.getSiteFileModDisableReason( selectedSite, 'modifyFiles' );
+
+	if ( reasons && reasons.length > 0 ) {
+		return [ 'calypso_plans_autoconfig_halt_filemod', { error: reasons[ 0 ] } ];
+	}
+
+	if ( ! selectedSite.hasMinimumJetpackVersion ) {
+		return [
+			'calypso_plans_autoconfig_halt_jpversion',
+			{ jetpack_version: selectedSite.options.jetpack_version },
+		];
 	}
 
-	return <PurchaseDetail { ...props } />;
+	if ( selectedSite.is_multisite && ! selectedSite.isMainNetworkSite ) {
+		return [ 'calypso_plans_autoconfig_halt_multisite', undefined ];
+	}
+
+	if ( selectedSite.options.is_multi_network ) {
+		return [ 'calypso_plans_autoconfig_halt_multinetwork', undefined ];
+	}
+
+	return null;
 };
 
-export default JetpackPlanDetails;
+const mapDispatchToProps = ( dispatch, { selectedSite } ) => ( {
+	trackAutoconfigHalt: () => {
+		const eventData = getTracksDataForAutoconfigHalt( selectedSite );
+
+		if ( ! eventData ) {
+			return;
+		}
+
+		const [ name, properties ] = eventData;
```

Comment:
My personal preference is to use objects instead of arrays in these cases, because they're more readable and expressive.

Reply:
done in 2ec88af712